state-chain/runtime/src/chainflip/bitcoin_block_processor.rs

```diff
@@ -72,24 +94,57 @@ impl Hook<(BlockNumber, BtcEvent), ()> for ExecuteEventHook {
 	Default,
 )]
 pub struct ApplyRulesHook {}
-impl Hook<(BlockNumber, BlockNumber, BlockData), Vec<(BlockNumber, BtcEvent)>> for ApplyRulesHook {
+impl
+	Hook<
+		(BlockNumber, BlockNumber, BlockData),
+		Vec<(BlockNumber, BtcEvent<DepositWitness<Bitcoin>>)>,
+	> for ApplyRulesHook
+{
 	fn run(
 		&mut self,
 		(block, age, block_data): (BlockNumber, BlockNumber, BlockData),
-	) -> Vec<(BlockNumber, BtcEvent)> {
+	) -> Vec<(BlockNumber, BtcEvent<DepositWitness<Bitcoin>>)> {
 		// Prewitness rule
 		if age == 0 {
 			return block_data
 				.iter()
 				.map(|deposit_witness| (block, BtcEvent::PreWitness(deposit_witness.clone())))
-				.collect::<Vec<(BlockNumber, BtcEvent)>>();
+				.collect::<Vec<_>>();
 		}
 		//Full witness rule
 		if age == BitcoinIngressEgress::witness_safety_margin().unwrap() {
 			return block_data
 				.iter()
 				.map(|deposit_witness| (block, BtcEvent::Witness(deposit_witness.clone())))
-				.collect::<Vec<(BlockNumber, BtcEvent)>>();
+				.collect::<Vec<_>>();
+		}
+		vec![]
+	}
+}
+
+impl
+	Hook<
+		(BlockNumber, BlockNumber, BlockDataVaultDeposit),
+		Vec<(BlockNumber, BtcEvent<VaultDepositWitness<Runtime, BitcoinInstance>>)>,
+	> for ApplyRulesHook
+{
+	fn run(
+		&mut self,
+		(block, age, block_data): (BlockNumber, BlockNumber, BlockDataVaultDeposit),
+	) -> Vec<(BlockNumber, BtcEvent<VaultDepositWitness<Runtime, BitcoinInstance>>)> {
+		// Prewitness rule
+		if age == 0 {
+			return block_data
+				.iter()
+				.map(|deposit_witness| (block, BtcEvent::PreWitness(deposit_witness.clone())))
+				.collect::<Vec<_>>();
+		}
+		//Full witness rule
+		if age == BitcoinIngressEgress::witness_safety_margin().unwrap() {
```

Comment:
I think it reads clearer like, splitting the ifs reads at first glance as though it's possible to reach both conditions. Using else if and else makes it structurally clear that only one of these branches will be hit.

```rust
		if age == 0 {
			block_data
				.iter()
				.map(|deposit_witness| (block, BtcEvent::PreWitness(deposit_witness.clone())))
				.collect::<Vec<_>>()
		} else if age == BitcoinIngressEgress::witness_safety_margin().unwrap() {
			block_data
				.iter()
				.map(|deposit_witness| (block, BtcEvent::Witness(deposit_witness.clone())))
				.collect::<Vec<_>>()
		} else {
			vec![]
		}
```

Reply:
same above too ofc

Reply:
resolving this after the discussion - we can look again later